File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/c++/write.cc
#include "write.h"
#include <iostream>
using namespace std;

void Write::ejecutar(Programa* programa){
    stack<int>& pila = programa->getPila();
    if (!pila.empty()) {
        cout << "= " << pila.top() << endl;
        pila.pop();
    } else {
        cout << "No hay nada en la pila." << endl;
    }
    programa->inc();
}

string Write::mostrar() {
    return "write";
}

File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/java/Write.java
import java.util.Stack;

public class Write extends Instruccion{

    public void ejecutar(Programa programa){
        Stack<Integer> pila = programa.getPila();
        if (!pila.isEmpty()){
            System.out.println("= " + pila.pop());
        } else{
            System.out.println("No hay nada en la pila.");
        }
        programa.inc();
    }

    public String mostrar(){
        return "write";
    }
}
